### PR TITLE
官方首页的搜索框，当按下键盘s的时候，会获取焦点

### PR DESCRIPTION
**Pull request by Manweill**
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [ ] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [ ] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [ ] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [ ] Rebase before creating a PR to keep commit history clear.
* [ ] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

### CODE REVIEW

File: site/theme/template/Layout/Header.jsx [hunk 1]
```diff
@@ -21,11 +21,24 @@ export default class Header extends React.Component {
...
+        focused: false,
+        inputValue: '',
+      });
+    });
+    searchInput.addEventListener('focus', () => {
+      this.setState({
+        focused: false,
+        inputValue: '',
+      });
+    });
```
Comment (afc163) [inline site/theme/template/Layout/Header.jsx]: Please follow React's way.

Comment (Manweill) [inline site/theme/template/Layout/Header.jsx]: 要用React的方式实现？

Comment (afc163) [inline site/theme/template/Layout/Header.jsx]: 是的

Comment (Manweill) [inline site/theme/template/Layout/Header.jsx]: 无论是AutoComplete还是Select都无法实现。<AutoComplete><Input ref={this.ref=ref}/></AutoComplete>这种形式是可以取到ref，但是会提示this.ref.focus()不是一个方法

Comment (benjycui) [inline site/theme/template/Layout/Header.jsx]: @Manweill Input 是有 [.focus](https://github.com/ant-design/ant-design/blob/293d63910a6c4ce11e743726063555d790b741c9/components/input/Input.tsx#L95-L97) 的，你再试试，如果碰到问题，可以提供重现 demo。

务必 think in react，谢谢、

Comment (afc163) [inline site/theme/template/Layout/Header.jsx]: 可能有这个问题 https://github.com/ant-design/ant-design/issues/6814

Comment (Manweill) [inline site/theme/template/Layout/Header.jsx]: 单独用Input的话，focus()是没问题，放到AutoComplete就会有问题了。重现的demo，我今晚放上来吧，或者你可以自己写一下试试。环境用dva初始化就可以

Comment (benjycui) [inline site/theme/template/Layout/Header.jsx]: > 可能有这个问题 #6814

@Manweill 的确是这个问题。。。

**Comment by benjycui** [comment]
@Manweill trace: https://github.com/ant-design/ant-design/pull/7237

**Comment by Manweill** [comment]
@benjycui 等官方修复AutoComplete吗

**Comment by benjycui** [comment]
@Manweill 已经修复，麻烦 rebase，然后你就可以用 ref 去 focus 了。

**Comment by Manweill** [comment]
@benjycui 好，我试试

**Pull request closed without merging**